### PR TITLE
Mbarba/cleanup

### PR DESCRIPTION
**Pull request by MatBarba**
Tweaks and fixes to make mypy happy

### CODE REVIEW

File: src/python/ensembl/io/genomio/utils/archive_utils.py [hunk 1]
```diff
@@ -16,14 +16,27 @@
...
 
 
+@contextmanager
+def open_gz_file(file_path: PathLike) -> Generator[TextIO, None, None]:
+    this_file = Path(file_path)
+    if this_file.suffix == ".gz":
+        with gzip.open(this_file, "rt") as fh:
+            yield fh
+    else:
+        with open(this_file, "rt") as fh:
```
Comment (JAlvarezJarreta) [inline src/python/ensembl/io/genomio/utils/archive_utils.py:36]:
```suggestion
        with this_file.open("rt") as fh:
```
To follow with other places in the codebase where it is used this way with `Path` variables.

File: src/python/tests/metadata/test_prepare_genome.py [hunk 2]
```diff
@@ -35,10 +35,13 @@
 
 
 class TestSchemas:
+    test_data_dir = Path()
+    tmp_dir = Path()
```
Comment (JAlvarezJarreta) [inline src/python/tests/metadata/test_prepare_genome.py:39]:
```suggestion
    test_data_dir: Path
    tmp_dir: Path
```
Although not 100%, it seems based on the information I have been able to find that this should be ok :crossed_fingers:

File: src/python/tests/test_schemas.py [hunk 3]
```diff
@@ -36,11 +36,15 @@
 
 
 class TestSchemas:
+    test_data_dir = Path()
+    schema_dir = Path()
+    tmp_dir = Path()
```
Comment (JAlvarezJarreta) [inline src/python/tests/test_schemas.py:41]:
```suggestion
    test_data_dir: Path
    schema_dir: Path
    tmp_dir: Path
```
Same as before..

**Review by JAlvarezJarreta — APPROVED** [review]
Great work! I know it is not something too enjoyable to make `mypy` or `pylint` happy, but at least it is worth the effort in the long run.

Just three minor comments, two of which are more a matter of preference than anything else (the last two).

**Review by vsitnik — APPROVED** [review]
cool. thanks!

File: src/python/ensembl/io/genomio/utils/archive_utils.py [hunk 4]
```diff
@@ -16,14 +16,27 @@
...
 from os import PathLike
 from pathlib import Path
 import shutil
+from typing import Generator, TextIO
 
 import argschema
 
 
+@contextmanager
+def open_gz_file(file_path: PathLike) -> Generator[TextIO, None, None]:
```
Comment (vsitnik) [inline src/python/ensembl/io/genomio/utils/archive_utils.py:30]: are we sure we don't want to pass additional args in `*args` and  have `"rt"`  hardcoded?

**Pull request closed without merging**